src/main/java/org/dsa/iot/calendar/event/EventUtils.java

```diff
@@ -0,0 +1,57 @@
+package org.dsa.iot.calendar.event;
+
+import java.time.*;
+import java.time.temporal.ChronoUnit;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.util.stream.Collectors.toList;
+
+public class EventUtils {
+    protected static Clock clock = Clock.systemDefaultZone();
+
+    public static TimeRange findNextFreeTimeRange(List<DSAEvent> events, Duration wantedDuration) {
+        Instant now = Instant.now(clock);
+        if (events.isEmpty()) {
+            return new TimeRange(now, now.plus(wantedDuration));
+        }
+
+        List<DSAEvent> eventsNotOver = events
+                .stream()
+                .filter(event -> event.getEnd().isAfter(Instant.now(clock)))
+                .sorted((o1, o2) -> {
+                    if (MINUTES.between(o1.getStart(), now) > MINUTES.between(o2.getStart(), now)) {
+                        return 1;
+                    } else if (MINUTES.between(o1.getStart(), now) < MINUTES.between(o2.getStart(), now)) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                }).collect(toList());
+
+        for (int i = 0; i < events.size() - 1; ++i) {
```

Comment:
Yes most likely.  This method was kind of on hold though.  Needed to get the time slot table done to finish it.